os/kernel/binary_manager/binary_manager_bootparam.c

```diff
@@ -0,0 +1,147 @@
+/****************************************************************************
+ *
+ * Copyright 2021 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <debug.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdbool.h>
+
+#include <tinyara/binary_manager.h>
+
+#include "binary_manager.h"
+
+/* Data for Boot parameters */
+static binmgr_bpinfo_t bp_info;
+
+#define BP_SEEK_OFFSET(index)   (index == 0 ? 0 : BOOTPARAM_PARTSIZE / 2)
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+/****************************************************************************
+ * Name: binary_manager_register_bppart
+ *
+ * Description:
+ *	 This function registers the partition of boot parameters.
+ *
+ ****************************************************************************/
+void binary_manager_register_bppart(int part_num, int part_size)
+{
+	if (part_num < 0 || part_size != BOOTPARAM_PARTSIZE) {
```

Comment:
If this function works only when `part_size` is same as `BOOTPARAM_PARTSIZE`, passing the size is needed?
And boot param partition size is checked when build time through `mkbootparam.py`.
So I think checking the `part_size` is not needed.

Reply:
What if part_num = 0? I don't think this case is successful.

Reply:
Yes, but I added this to check old binary which bootparam is not applied.